Amplify/Categories/DataStore/Model/Internal/Schema/AuthRule.swift

```diff
@@ -22,6 +22,12 @@ public enum ModelOperation {
     case update
     case delete
     case read
+    // granular read access
+    case `get`
+    case list 
+    case sync 
+    case listen 
+    case search
 }
```

Comment:
Can we start off by attaching the generated Model swift types that utilizes these new granular read access cases? By doing so, we can update the source schema that we use to reference the generated types. 

Is there an sort of relevant API calls that we can write, either in the unit or integration tests, utilizing these models, and asserting the expected behavior?

Reply:
The example schema using it is 
```
type Todo @model
@auth(rules:[
  { allow: public, operations: [get, list, listen, sync, search] },
  { allow: owner }
])
{
  id: ID!
  name: String
}
```
The model files can be generated by the current version of modelgen, which has the following line generated
```
    model.authRules = [
      rule(allow: .public, operations: [.get, .list, .listen, .sync, .search]),
      rule(allow: .owner, ownerField: "owner", identityClaim: "cognito:username", provider: .userPools, operations: [.create, .update, .delete, .read])
    ]
```